src/helpers/checkForProblematicUnflatKeys.ts

```diff
@@ -0,0 +1,24 @@
+import { messages } from '../messages';
+import { getLogger } from '../helpers/logger';
+
+export function checkForProblematicUnflatKeys(obj: object) {
+  const allKeys = Object.keys(obj).sort();
+  const problematicKeys = [];
+  const max = allKeys.length - 1;
+  for (let i = 0; i < max; ) {
+    const key = allKeys[i];
+    const prefix = key + '.';
+    let index = allKeys[++i].indexOf(prefix);
```

Comment:
Use `startsWith` instead of `indexOf`, try to give `index` a more explanatory name (maybe `nextKeyIsNested`? or anything that implies what we are checking)